### PR TITLE
HDDS-9491. Migrate tests in hdds to JUnit5

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
This change is to migrate tests in hdds to JUnit5. Tests migrated, are listed below.
```
hadoop-hdds/client/src/test/java/org/apache/hadoop/hdds/scm/TestContainerClientMetrics.java
hadoop-hdds/client/src/test/java/org/apache/hadoop/hdds/scm/storage/TestBlockOutputStreamCorrectness.java
hadoop-hdds/client/src/test/java/org/apache/hadoop/hdds/scm/storage/TestChunkInputStream.java
hadoop-hdds/common/src/test/java/org/apache/hadoop/hdds/utils/db/Proto2CodecTestBase.java
hadoop-hdds/config/src/test/java/org/apache/hadoop/hdds/conf/TestConfigFileAppender.java
hadoop-hdds/config/src/test/java/org/apache/hadoop/hdds/conf/TestConfigFileGenerator.java
hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/ssl/TestPemFileBasedKeyStoresFactory.java
hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/ssl/TestReloadingX509KeyManager.java
hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/ssl/TestReloadingX509TrustManager.java
hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/token/TestOzoneBlockTokenSecretManager.java
hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/server/TestServerUtils.java
hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/server/http/TestHttpServer2MetricsTest.java
hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/utils/db/TestFixedLengthStringCodec.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/balancer/TestMoveManager.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/TestOverReplicatedProcessor.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/TestRatisOverReplicationHandler.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/TestRatisUnderReplicationHandler.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/TestReplicationManager.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/TestReplicationManagerUtil.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/TestUnderReplicatedProcessor.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/health/TestECMisReplicationCheckHandler.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/TestInterSCMGrpcProtocolService.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/upgrade/TestScmFinalization.java
hadoop-hdds/tools/src/test/java/org/apache/hadoop/hdds/scm/cli/container/upgrade/TestUpgradeManager.java
```

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9491

## How was this patch tested?
CD/CI run on the fork: https://github.com/hemantk-12/ozone/actions/runs/6961062933

### CODE REVIEW

File: hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/utils/db/TestRDBTableStore.java [hunk 1]
```diff
@@ -737,9 +736,9 @@ public void testDumpAndLoadEmpty() throws Exception {
       testTable1.dumpToFileWithPrefix(dumpFile, samplePrefix);
 
       // check dump file exist
-      Assert.assertTrue(dumpFile.exists());
+      assertTrue(dumpFile.exists());
       // empty dump file
-      Assert.assertTrue(dumpFile.length() == 0);
+      Assertions.assertEquals(0, dumpFile.length());
```
Comment (adoroszlai) [inline hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/utils/db/TestRDBTableStore.java:741]: Nit:

```suggestion
      assertEquals(0, dumpFile.length());
```

File: hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/token/TestOzoneBlockTokenSecretManager.java [hunk 2]
```diff
@@ -104,30 +106,30 @@ public void testGenerateToken() throws Exception {
...
+    assertEquals(BlockTokenVerifier.getTokenService(blockID),
         btIdentifier.getService());
-    Assert.assertEquals(EnumSet.allOf(AccessModeProto.class),
+    assertEquals(EnumSet.allOf(AccessModeProto.class),
         btIdentifier.getAccessModes());
     byte[] hash = secretManager.createPassword(btIdentifier);
-    Assert.assertEquals(secretKeyId, btIdentifier.getSecretKeyId());
-    validateHash(hash, btIdentifier.getBytes());
+    assertEquals(secretKeyId, btIdentifier.getSecretKeyId());
+    assertFalse(secretKey.isValidSignature(hash, btIdentifier.getBytes()));
```
Comment (adoroszlai) [inline hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/token/TestOzoneBlockTokenSecretManager.java:132]: Same here:

```suggestion
    assertTrue(secretKey.isValidSignature(btIdentifier.getBytes(), hash));
```

File: hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/token/TestOzoneBlockTokenSecretManager.java [hunk 3]
```diff
@@ -50,8 +49,11 @@
 import static org.apache.hadoop.ozone.container.ContainerTestHelper.getReadChunkRequest;
 import static org.apache.hadoop.ozone.container.ContainerTestHelper.newPutBlockRequestBuilder;
 import static org.apache.hadoop.ozone.container.ContainerTestHelper.newWriteChunkRequestBuilder;
-import static org.junit.Assert.assertThrows;
-import static org.junit.Assert.assertTrue;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
Comment (adoroszlai) [inline hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/token/TestOzoneBlockTokenSecretManager.java:56]: Import tweak for the signature validation change:

```suggestion
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;
```

**Review by adoroszlai — COMMENTED** [review]
Thanks a lot @hemantk-12 for taking up this task, and going even further, like switching to `assertThat(..., containsString())`.

File: hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/token/TestOzoneBlockTokenSecretManager.java [hunk 4]
```diff
@@ -104,30 +106,30 @@ public void testGenerateToken() throws Exception {
...
         identifier.getService());
-    Assert.assertEquals(EnumSet.allOf(AccessModeProto.class),
+    assertEquals(EnumSet.allOf(AccessModeProto.class),
         identifier.getAccessModes());
-    Assert.assertEquals(secretKeyId, identifier.getSecretKeyId());
-
-    validateHash(token.getPassword(), token.getIdentifier());
+    assertEquals(secretKeyId, identifier.getSecretKeyId());
+    assertFalse(secretKey.isValidSignature(token.getPassword(),
+        token.getIdentifier()));
```
[the remark this answers is not in the document]
Comment (hemantk-12) [reply hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/token/TestOzoneBlockTokenSecretManager.java:115]: Thanks for the catch.

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @hemantk-12 for the quick turnaround.

**Pull request merged**